tests/integration/ha_tests/helpers.py

```diff
@@ -686,11 +701,11 @@ async def send_signal_to_process(
 
             if signal not in ["SIGSTOP", "SIGCONT"]:
                 _, raw_pid, _ = await ops_test.juju(
-                    "ssh", "--container", "postgresql", unit_name, "pgrep", "-x", process
+                    "ssh", "--container", "postgresql", unit_name, "pgrep", opt, process
                 )
 
-                # If something was returned, there is a running process.
-                if len(raw_pid) > 0:
+                # If the same output was returned, process was not restarted.
+                if raw_pid == old_pid:
```

Comment:
Pebble might have restarted the process by the time we ask, so we might receive different pids.

Reply:
Great idea.